torch/_dynamo/eval_frame.py

```diff
@@ -360,9 +359,79 @@ def first_real_inst_idx(code):
     raise RuntimeError("RESUME instruction not found in code")
 
 
+# Converts a CodePart structure, which carries our
+# re-compilation code, source, and more, into a
+# plan for how to mark certain dims as dynamic.
+def dynamic_plan_from_code_part(code_part, dynamic_plan):
```

Comment:
Some type annotations here would be extremely helpful, since both of these quantities come from arguments to the callback

Reply:
sg